tests/streams/test_cache.py

```diff
@@ -1,299 +0,0 @@
-        {  # missing lastest index
-            "id": "parent-uuid",
-            "latest_id": "record-uuid",
-        },
-        {  # no latest id nor next draft id
-            "id": "parent-uuid",
-            "latest_index": 1,
-        },
-    ]
-
-    for entry in invalid:
-        pytest.raises(AssertionError, parents_state.add, "123", entry)
-
-
-###
-# Record State
-###
-
-
-@pytest.fixture(scope="function")
-def records_state():
-    """Records state."""
-    return RecordsState()
-
-
-def test_record_state_record_entry(records_state):
-    records_state.add(
-        "123",
-        {
-            "index": 1,
-            "id": "record-uuid",
-            "parent_id": "parent-uuid",
-            "fork_version_id": 1,
-        },
-    )
-    assert records_state.get("123")
-    assert records_state.get(123)  # test num to string conversion
-
-
-def test_record_state_invalid_entries(records_state):
-    invalid = [
-        {
-            "id": "record-uuid",
-            "parent_id": "parent-uuid",
-            "fork_version_id": 1,
-        },
-        {
-            "index": 1,
-            "parent_id": "parent-uuid",
-            "fork_version_id": 1,
-        },
-        {
-            "index": 1,
-            "id": "record-uuid",
-            "fork_version_id": 1,
-        },
-        {
-            "index": 1,
-            "id": "record-uuid",
-            "parent_id": "parent-uuid",
-        },
-    ]
-
-    for entry in invalid:
-        pytest.raises(AssertionError, records_state.add, "123", entry)
-
-
-###
-# State
-###
-
-
-class MockState(State):
-    """Mock State."""
-
-    def _validate(self, data):
-        """Validate data value is not 2."""
-        assert not data["value"] == 2
-
-
-@pytest.fixture(scope="function")
-def data_file(tmp_dir):
-    """State data filepath."""
-    filepath = Path(tmp_dir.name) / "data.json"
-    with open(filepath, "w") as file:
-        file.write(json.dumps({"one": {"value": 1}, "two": {"value": 2}}))
-
-    return filepath
-
-
-def test_state_from_source_data_no_validation(data_file):
-    """Test creating a state with initial data."""
-    state = MockState(data_file)
-
-    assert state.get("one")
-    assert state.get("two")
-    assert len(state.all()) == 2
-
-
-def test_state_from_source_data_with_validation(data_file):
-    """Test creating a state with initial data."""
-    with pytest.raises(AssertionError):
-        _ = MockState(data_file, validate=True)
-
-
-def test_state_dump_data(tmp_dir):
-    """Test dumping the data of the state."""
-    state = MockState()
-
-    state.add("one", {"value": 1})
-    state.add("three", {"value": 3})
-    state.add("four", {"value": 4})
-
-    filepath = Path(tmp_dir.name) / "dump.json"
-    state.dump(filepath)
-
-    with open(filepath, "r") as file:
-        expected = '{"one": {"value": 1}, "three": {"value": 3}, "four": {"value": 4}}'
-        content = file.read()
-        assert content == expected
-
-
-###
-# PIDMaxPK State
```

Comment:
max pid pk cache/state tests are not needed anymore since validation happens on the db level by column typing.